HelpSource/scdoc.js

```diff
@@ -282,3 +282,17 @@ function setUpWebChannel(port) {
         });
     }
 }
+
+function renderTex() {
+    for (let aSpan of document.getElementsByClassName("math")) {
+        const tex = aSpan.textContent;
+        katex.render(tex, aSpan, {
+            throwOnError: false,
+            displayMode: aSpan.classList.contains("block"),
+        });
+    }
+}
+
+window.addEventListener('DOMContentLoaded',function () {
```

Comment:
I'm not super familiar with web tech, but [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) says 

'DOMContentLoaded may fire before your script has a chance to run, so it is wise to check before adding a listener.'

...and gives this solution...
```js
function doSomething() {
  console.info("DOM loaded");
}

if (document.readyState === "loading") {
  // Loading hasn't finished yet
  document.addEventListener("DOMContentLoaded", doSomething);
} else {
  // `DOMContentLoaded` has already fired
  doSomething();
}
```

Is that something that should be done here?

Reply:
MDN says

> The DOMContentLoaded event fires when the HTML document has been completely parsed, and all deferred scripts ([<script defer src="…">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer) and [<script type="module">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#module)) have downloaded and executed. It doesn't wait for other things like images, subframes, and async scripts to finish loading.

which is pertty much what we want here as we work with a defered script and need to wait for the DOM to be available in the browser. I think I also copied this code mostly from the kaTeX docs https://katex.org/docs/autorender

I think everything is as it should be regarding the execution of JS code.